Sources/Basic/FileSystem.swift

```diff
@@ -500,7 +494,60 @@ public struct RerootedFileSystemView: FileSystem {
     public mutating func writeFileContents(_ path: AbsolutePath, bytes: ByteString) throws {
         return try underlyingFileSystem.writeFileContents(formUnderlyingPath(path), bytes: bytes)
     }
+
+    public func openFileOutputStream(_ path: AbsolutePath) throws -> FileOutputByteStream {
+        return try underlyingFileSystem.openFileOutputStream(path)
+    }
 }
 
 /// Public access to the local FS proxy.
 public var localFileSystem: FileSystem = LocalFileSystem()
+
+/// Implements file output stream for InMemory file system.
+private final class InMemoryFileOutputByteStream: FileOutputByteStream {
+
+    /// Reference to the file system.
+    private let fileSystem: InMemoryFileSystem
+
+    /// Path of the file.
+    private let path: AbsolutePath
+
+    private var error: Swift.Error?
+
+    // FIXME: Need to check and throw error if file can't be written at provided path.
+    /// Creates a stream for the provided path in a in memory file system.
+    init(fileSystem: InMemoryFileSystem, path: AbsolutePath) throws {
+        self.fileSystem = fileSystem
+        self.path = path
+        super.init()
+    }
+
+    override func writeImpl(_ bytes: [UInt8]) {
+        // FIXME: This is currently highly inefficient. This reads the contents of file if present and re-writes.
+        // We need append file contents semantics in InMemoryFileSystem to make this efficient.
```

Comment:
I would expect this implementation to simply fully buffer the file before writing.

That makes its honoring of the `flush` semantics a little suspect (as it won't immediately write to disk, or if it does it is going to be inefficient), but I'd prefer that as a starting point.